traffic_portal/test/integration/PageObjects/UsersPage.po.ts

```diff
@@ -61,22 +109,94 @@ export class UsersPage extends BasePage {
       await this.btnCreateNewUser.click();
       await this.txtFullName.sendKeys(user.FullName + this.randomize);
       await this.txtUserName.sendKeys(user.Username + this.randomize);
-      await this.txtEmail.sendKeys(user.FullName + this.randomize + user.Email);
+      await this.txtEmail.sendKeys(user.Email + this.randomize);
       await this.txtRole.sendKeys(user.Role);
       await this.txtTenant.sendKeys(user.Tenant+this.randomize);
       await this.txtPassword.sendKeys(user.Password);
       await this.txtConfirmPassword.sendKeys(user.ConfirmPassword);
       await this.txtPublicSSHKey.sendKeys(user.PublicSSHKey);
       await basePage.ClickCreate();
-      if(await basePage.GetOutputMessage() == user.existsMessage){
+      if(await basePage.GetOutputMessage() === user.existsMessage){
         await snp.NavigateToUsersPage();
         result = true;
-      }else if(await basePage.GetOutputMessage() == user.validationMessage){
+      }else if(await basePage.GetOutputMessage() === user.validationMessage){
         result = true;
       }else{
         result = false;
       }
       return result;
     }
 
+    public async SearchUser(nameUser: string) {
+        let snp = new SideNavigationPage();
+        let name = nameUser + this.randomize;
+        await snp.NavigateToUsersPage();
+        await this.txtSearch.clear();
+        await this.txtSearch.sendKeys(name);
+        await element.all(by.repeater('u in ::users')).filter(function (row) {
+            return row.element(by.name('username')).getText().then(function (val) {
+                return val === name;
+            });
+        }).first().click();
+    }
+
+    public async SearchEmailUser(nameEmail: string) {
+        let snp = new SideNavigationPage();
+        let name = nameEmail + this.randomize;
+        await snp.NavigateToUsersPage();
+        await this.txtSearch.clear();
+        await this.txtSearch.sendKeys(name);
+        await element.all(by.repeater('u in ::users')).filter(function (row) {
+            return row.element(by.name('email')).getText().then(function (val) {
+                return val === name;
+            });
+        }).first().click();
+    }
+
+    public async UpdateUser(user: UpdateUser): Promise<boolean  | undefined> {
```

Comment:
It looks like the tests actually only check that the resolution of this is "truthy" or literally `true`, so why not just return a `Promise<boolean>` instead of allowing it to be `undefined`? Allowing `undefined` could be confusing, for example, if `UpdateUser` returns a `Promise<boolean>`, these are the same check:

```typescript
expect(await usersPage.UpdateUser(u)).toBeTrue(); // or .toBe(true)
expect(await usersPage.UpdateUser(u)).not.toBeFalse(); // or .toBe(false)
```

... but if it returns a `Promise<boolean | undefined>`, they are **not** the same. These pairs of checks also have the same problem where they're identical for `Promise<boolean>` but not `Promise<boolean>`:

```typescript
expect(await usersPage.UpdateUser(u)).toBeFalse();
expect(await usersPage.UpdateUser(u)).toBe(false);
expect(await usersPage.UpdateUser(u)).toBeFalsey()
```